### DISCUSSION TITLE
Add outside-collaborators to internal forks.

### DISCUSSION THREAD

**Discussion by JeroenKnoops** [mesh-iit/outside-collaborators#161] [Q&A] [answered]
## Use-case
GitHub has released a features which allows enterprise to fork internally.

This is great, because now we can fork a repository and add specific outside-collaborators to the fork only, shielding off the internal conversations and only sync major releases to the downstream.

## The Problem
Unfortunately GitHub does not allow you to add outside-collaborators to internal forks. :(

### Error
You will get the following error:

```
- Inviting collaborator "my-oc" with permissions "read" - problem detected: #<Octokit::UnprocessableEntity: PUT https://api.github.com/repos/my-organization/my-internally-forked-repository-name/collaborators/my-oc: 422 - Validation Failed
Error summary:
  resource: Repository
  code: custom
  message: Users outside of the enterprise account cannot be invited // See: https://docs.github.com/rest/reference/repos#add-a-repository-collaborator> ❌
```

**Comment by JeroenKnoops** [comment, marked as answer]
## Workaround
Also grant access to the upstream. When this invitation is accepted, the downstream / fork will also be accepted.

**Reply by pattacini** [reply]
Thanks @JeroenKnoops for sharing this use case and a possible workaround 👍🏻 

I really look forward to having internal forks in GA soon.
